bot/cogs/metrics.py

```diff
@@ -0,0 +1,77 @@
+from collections import defaultdict
+
+from discord import Member, Message
+from discord.ext.commands import Cog
+from prometheus_client import Counter, Gauge
+
+from bot.bot import Bot
+
+
+class Metrics(Cog):
+    """
+    Exports metrics for Prometheus.
+
+    See https://github.com/prometheus/client_python for metric documentation.
+    """
+
+    PREFIX = 'pydis_bot_'
+
+    def __init__(self, bot: Bot) -> None:
+        self.bot = bot
+
+        self.guild_members = Gauge(
+            name=f'{self.PREFIX}guild_members',
```

Comment:
I think it'd be cleaner if `PREFIX` didn't have a trailing underscore and the underscore was instead part of this fstring.

Reply:
see 4dd4efc